x-pack/plugins/reporting/export_types/printable_pdf/server/lib/layouts/preserve_layout.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import path from 'path';
+import {
+  kbn_server,
+  ViewWidthHeight,
+  ViewZoomWidthHeight,
+} from '../../../../../../../../src/server/index';
+import { Layout } from './layout';
+
+interface Pagesizeparams {
+  pageMarginTop: number;
+  pageMarginBottom: number;
+  pageMarginWidth: number;
+  tableBorderWidth: number;
+  headingHeight: number;
+  subheadingHeight: number;
+}
+
+export class Preservelayout extends Layout {
+  public groupCount: number = 1;
+
+  public selectors = {
+    screenshot: '[data-shared-items-container]',
+    renderComplete: '[data-shared-item]',
+    itemsCountAttribute: 'data-shared-items-count',
+    timefilterFromAttribute: 'data-shared-timefilter-from',
+    timefilterToAttribute: 'data-shared-timefilter-to',
+  };
+
+  public height: number = 0;
+  public width: number = 0;
+  public zoom: number = 0;
```

Comment:
We still need to set a zoom but since it's not configurable for Preserve Layout, I would just make this a const and move it outside the class:

const ZOOM = 2; 

type is inferred so shouldn't need to specify number here.